openedx/features/course_experience/api/v1/serializers.py

```diff
@@ -1,8 +1,20 @@
 """
 Serializer for Course Deadlines (Mobile)
 """
+from rest_framework import serializers
+
+from opaque_keys.edx.keys import CourseKey
+
 from lms.djangoapps.course_home_api.mixins import DatesBannerSerializerMixin
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
 
 class CourseDeadlinesMobileSerializer(DatesBannerSerializerMixin):
-    pass
+    has_ended = serializers.SerializerMethodField()
```

Comment:
Should this just live inside the DatesBannerSerializerMixin?

And out of curiosity, why move all the logic into the serializer vs the view? This also stopped the check of the get_course_with_access which could return a no access for a user, or does the CourseOverview.get_from_id have that implicit?